Xamarin.Forms.Core/BindingExpression.cs

```diff
@@ -371,6 +372,16 @@ bool TryConvert(BindingExpressionPart part, ref object value, Type convertTo, bo
 			object original = value;
 			try
 			{
+				var stringValue = value as string ?? string.Empty;
+				// see: https://bugzilla.xamarin.com/show_bug.cgi?id=32871
+				// do not canonicalize "*.[.]"; "1." should not update bound BindableProperty
+				if (stringValue.EndsWith(".") && DecimalTypes.Contains(convertTo))
```

Comment:
Value is not modified until the end, so why not simply return false instead of throwing?